pycatchmod/__main__.py

```diff
@@ -28,12 +28,24 @@ def compare(ctx, filename, plot):
     from pycatchmod.io.excel import compare
     compare(filename, plot)
 
+def date_parser(x):
+    import pandas
+    # try creating a date range based on the first date and the length
+    dates = pandas.period_range(x[0], periods=len(x), freq="D")
+    if pandas.Period(x.values[-1]) != dates[-1]:
+        # parse dates one by one (much slower)
+        # note this will assume american-style dates when faced with ambiguity
+        dates = pandas.PeriodIndex(x.values, freq="D")
+    return dates
+
 def pandas_read(filename, key=None):
     import pandas  # TODO: move this somewhere else?
     if filename.endswith((".h5", ".hdf", ".hdf5")):
         df = pandas.read_hdf(filename, key=key)
     elif filename.endswith((".csv")):
-        df = pandas.read_csv(filename, parse_dates=True, dayfirst=True)
+        df = pandas.read_csv(filename)
```

Comment:
Yes. You can pass a date_parser to read_csv, but it still tries to convert to Timestamp.